src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs

```diff
@@ -0,0 +1,309 @@
+        /// </summary>
+        public virtual MethodBase GetMethod()
+        {
+            return _method;
+        }
+
+        /// <summary>
+        /// Returns the offset from the start of the native (jitted) code for the
+        /// method being executed
+        /// </summary>
+        public virtual int GetNativeOffset()
+        {
+            return _nativeOffset;
+        }
+
+
+        /// <summary>
+        /// Returns the offset from the start of the IL code for the
+        /// method being executed.  This offset may be approximate depending
+        /// on whether the jitter is generating debuggable code or not.
+        /// </summary>
+        public virtual int GetILOffset()
+        {
+            return _ilOffset;
+        }
+
+        /// <summary>
+        /// Returns the file name containing the code being executed.  This
+        /// information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual string GetFileName()
+        {
+            return _fileName;
+        }
+
+        /// <summary>
+        /// Returns the line number in the file containing the code being executed.
+        /// This information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual int GetFileLineNumber()
+        {
+            return _lineNumber;
+        }
+
+        /// <summary>
+        /// Returns the column number in the line containing the code being executed.
+        /// This information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual int GetFileColumnNumber()
+        {
+            return _columnNumber;
+        }
+
+        /// <summary>
+        /// Builds a readable representation of the stack frame
+        /// </summary>
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder(255);
+#if !CORECLR
+            bool includeFileInfoIfAvailable;
+#endif
+
+            if (_method != null)
+            {
+                sb.Append(_method.Name);
+
+                // deal with the generic portion of the method
+                if (_method is MethodInfo methodInfo && methodInfo.IsGenericMethod)
+                {
+                    Type[] typars = methodInfo.GetGenericArguments();
+
+                    sb.Append('<');
+                    int k = 0;
+                    bool fFirstTyParam = true;
+                    while (k < typars.Length)
+                    {
+                        if (fFirstTyParam == false)
+                            sb.Append(',');
+                        else
+                            fFirstTyParam = false;
+
+                        sb.Append(typars[k].Name);
+                        k++;
+                    }
+
+                    sb.Append('>');
+                }
+#if !CORECLR
```

Comment:
Would it be better to add `AppendStackFrameWithoutMethodBase` method to the CoreCLR specific part that just returns false?

Reply:
OK. Thanks, will do that.